samples/snippets/src/test/java/com/example/dialogflow/CreateConversationProfileTest.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.dialogflow;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import com.google.cloud.dialogflow.v2.ConversationProfile;
+import com.google.cloud.dialogflow.v2.ConversationProfilesClient;
+import com.google.cloud.dialogflow.v2.HumanAgentAssistantConfig.SuggestionFeatureConfig;
+import com.google.cloud.dialogflow.v2.KnowledgeBaseName;
+import com.google.cloud.dialogflow.v2.SuggestionFeature.Type;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+@SuppressWarnings("checkstyle:abbreviationaswordinname")
+public class CreateConversationProfileTest {
+
+  private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static final String LOCATION = "global";
+  private static String conversationProfileNameToDelete = null;
+  private ByteArrayOutputStream bout;
+  private PrintStream out;
+
+  private static void requireEnvVar(String varName) {
+    assertNotNull(System.getenv(varName));
+  }
+
+  private static void deleteConversationProfile(String conversationProfileName) throws IOException {
+    try (ConversationProfilesClient conversationProfilesClient =
+        ConversationProfilesClient.create()) {
+      conversationProfilesClient.deleteConversationProfile(conversationProfileName);
+    }
+  }
+
+  @BeforeClass
+  public static void checkRequirements() {
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
```

Comment:
Probably shouldn't be a requirement these days, since we specifically recommend users don't use SA's locally.

Reply:
Yes it's kind of risky, but currently the sample code suggests it, should we keep the consistency for now or refactor a new authentication method for the current testing environment?
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md#testing-setup

Reply:
@kolea2 FYI - we should change our standard.